iocore/hostdb/HostDB.cc

```diff
@@ -2224,13 +2227,14 @@ put_hostinfo_ClusterFunction(ClusterHandler *ch, void *data, int /* len ATS_UNUS
 int
 HostDBContinuation::backgroundEvent(int /* event ATS_UNUSED */, Event * /* e ATS_UNUSED */)
 {
-  ++hostdb_current_interval;
+  // No nothing if hosts file checking is not enabled.
+  if (hostdb_hostfile_check_interval == 0) {
+    return EVENT_CONT;
```

Comment:
The previous logic was to return ``EVENT_CONT``. The background event is written to trigger every second but we only check the actual changes after the check interval. I assumed that it was done this way so that you can update the check interval without having to wait for it to expire.